azure-mgmt-batchai/src/main/java/com/microsoft/azure/management/batchai/implementation/AzureFileShareImpl.java

```diff
@@ -71,19 +66,19 @@ private AzureStorageCredentialsInfo ensureCredentials() {
     }
 
     @Override
-    public DefinitionStages.WithAttach<BatchAICluster.DefinitionStages.WithCreate> withKeyVaultSecretReference(KeyVaultSecretReference keyVaultSecretReference) {
+    public DefinitionStages.WithAttach<ParentT> withKeyVaultSecretReference(KeyVaultSecretReference keyVaultSecretReference) {
         ensureCredentials().withAccountKeySecretReference(keyVaultSecretReference);
         return this;
     }
 
     @Override
-    public DefinitionStages.WithAttach<BatchAICluster.DefinitionStages.WithCreate> withFileMode(String fileMode) {
+    public DefinitionStages.WithAttach<ParentT> withFileMode(String fileMode) {
         inner().withFileMode(fileMode);
         return this;
     }
 
     @Override
-    public DefinitionStages.WithAttach<BatchAICluster.DefinitionStages.WithCreate> withDirectoryMode(String directoryMode) {
+    public DefinitionStages.WithAttach<ParentT> withDirectoryMode(String directoryMode) {
         inner().withDirectoryMode(directoryMode);
         return this;
     }
```

Comment:
For all methods in this impl (AzureFileShareImpl) :  in Java fluent the return type of methods in `Impl` is not the interface members.  (e.g. instead of `DefinitionStages.WithAttach<ParentT> withDirectoryMode() ` it should be `AzureFileShareImpl withDirectoryMode()` any specific reason for doing this?

Reply:
It is a generic class implementing generic interface - if I change return type to ...Impl, compiler will show a warning.